pycbc/results/pygrb_postprocessing_utils.py

```diff
@@ -0,0 +1,868 @@
+# Copyright (C) 2019 Francesco Pannarale, Gino Contestabile, Cameron Mills
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+
+# =============================================================================
+# Preamble
+# =============================================================================
+
+"""
+Module to generate PyGRB figures: scatter plots and timeseries.
+"""
+
+import sys
+import glob
+import os
+import logging
+import argparse
+import copy
+import numpy
+from scipy import stats
+from pycbc.detector import Detector
+# TODO: imports to fix/remove
+try:
+    from glue import segments
+    from glue.ligolw import utils, lsctables, ligolw, table
+    from glue.segmentsUtils import fromsegwizard
+except ImportError:
+    pass
+try:
+    from pylal import MultiInspiralUtils
+except ImportError:
+    pass
```

Comment:
`glue` and `pylal` need replaced

Reply:
Here I switched all `glue` imports accordingly, but I am not sure I can do much about the `pylal` one.  We had encapsulated the `pylal` imports in the post-processing in `try...except` structures so the automatic tests would not be affected and with the plan to remove them as we developed things further.  This one up here is our last `pylal` import in the post-processing and there are no `MultiInspiralUtils` in `ligo` and its subpackages, so I would not know how to replace it.  Once we move to `hdf5`, this will become an obsolete dependency.

Btw, there are a couple more `pylal` imports in PyCBC.
1.  In `pycbc/results/legacy_grb.py`, but this file will be removed one the post-processing development is completed.  None of the new scripts or methods rely on it.
2. In `bin/pygrb/pycbc_make_offline_grb_workflow` when the post-processing is set up, and similarly to the above this part of the workflow will be soon superseded.
3. In `bin/pycbc_fit_sngl_trigs`.

@spxiwh, @ahnitz, can we leave the lines 42-45 @a-r-williamson quoted?  Ultimately they are already a narrowed down version of https://github.com/gwastro/pycbc/blob/master/pycbc/results/pygrb_plotting_utils.py#L38-L42 (and per this PR pygrb_plotting_utils.py will no longer have those lines).

Reply:
I didn't think pylal had ever been updated for python3, so is there ever a use case where this *can* do anything useful?

Reply:
Right now that would only be running with Python2.7 to reproduce old results.  That's a way of ensuring the post-processing scripts do not break things in terms of analyses.

Should I move https://github.com/gwastro/pycbc-pylal/blob/master/pylal/MultiInspiralUtils.py#L66 into `pygrb_postprocessing_utils.py`?  Well, after readapting it and removing parts that are not needed.

I can't think of any other solution.  Moving ahead with the xml->hdf5 work with this PR would just make it too big in my opinion.

Reply:
But PyCBC now does not work *at all* on python 2.7. Older versions of PyCBC are needed if trying to reproduce old results.

Reply:
I've commented on this in a summary message below (March 4, 2022) and we should continue the discussion from there.  In any event, all the code lines pointed out by @a-r-williamson are now updated.